### ISSUE TITLE
Interface for a user to add an item to be tracked

### ISSUE THREAD

**Issue by hahuang65** [BitSmithy/price_tracker#23]
(Ripped from #11, since that was re-purposed)
Once #1 is complete, we can start making an interface for anyone (user or not) to add an item to be tracked.

This should

Add an item if it's not already tracked
Do nothing / update the existing record if it's already being tracked
The reason I say "anyone" can add an item is because, I don't think tracking an item should be user-dependent.
An item should simply be tracked or not. It should not be tracked by a specific person.
A person can subscribe to a tracked item, but the tracked item itself should not be tied to a user.
